internal/server/polling/server.go

```diff
@@ -97,9 +99,86 @@ func (s *Server) poll(ctx context.Context, resource types.NamespacedName, secret
 		return fmt.Errorf("failed to list pull requests: %w", err)
 	}
 
+	// Log the PRs.
 	for _, pr := range prs {
 		s.log.Info("pull request", "pr", pr)
 	}
 
+	annotations := map[string]string{
+		bbp.AnnotationBBPKey: bbp.AnnotationBBPValue,
+	}
+
+	// List Terraform objects, created by BBP.
+	tfList, err := s.listTerraformObjects(ctx, tf, annotations)
```

Comment:
If you used a label for this purpose, you'd be able to use label selectors (the API server might even help out with indexing, and the controller-runtime cache certainly will).

Reply:
(You could consider using onwer references for this, too; but I understand if you want to restrict the scope of this PR).

Reply:
@yitsushi is there a reason you want to use annotations instead of labels for defining which objects are managed by BBP?

Reply:
Nothing particular.

Reply:
OK, then I can switch to using labels.